pages/[slug].js

```diff
@@ -0,0 +1,131 @@
+import { useRouter } from "next/router.js";
+import Image from "next/image";
+import surfspots from "../lib/surfspots";
+import styled from "styled-components";
+import Link from "next/link";
+import back_button from "../public/images/back_button.png";
+
+export default function SpotDetails() {
+  const router = useRouter();
+  const { slug } = router.query;
+
+  const currentSpot = surfspots.find((spot) => spot.slug === slug);
+
+  if (!currentSpot) {
+    return <p>Not found</p>;
+  }
+
+  return (
+    <>
+      <StyledImage>
+        <Image
+          src={currentSpot.image}
+          alt={`cover image of ${currentSpot.name}`}
+          width={800}
+          height={600}
+        />
+        <StyledTitle>{currentSpot.name}</StyledTitle>
+      </StyledImage>
+      <Link href="/">
+        <StyledBackButton
+          src={back_button}
+          alt="Back_Btn"
+          width={30}
+          height={30}
+        />
+      </Link>
```

Comment:
Thats not semantic unfortunately (https://caninclude.glitch.me/caninclude?child=button&parent=link)
But you can include the SVG instead :)